### DISCUSSION TITLE
Streaks aren't working for some habits (v 2.2.1)

### DISCUSSION THREAD

**Discussion by pseudacris** [iSoron/uhabits#2074] [Help & FAQ] [answered]
Some habits aren't creating streaks, such as a habit set for twice a month and another set for twice a week. The number of days meet the requirement I set but it's not counted as a streak, there are no empty checkmarks that fill in the gaps and I have to use skip days to fill in the gaps. 

Also, on another habit, I used a skip day for one day on a 3 day a week habit and then had to use consecutive skip days until the next day on the following week I performed it in order to maintain the streak. I'm not sure if that's how it's supposed to work or not. I performed that habit 2 out of the 3 times it was scheduled for and I assumed that one skip day would be enough? It's been a very long time since I've used skip days so I'm not sure if that's how it normally works.

**Comment by KyleSCraig** [comment]
The skip behavior sounds right to me, you're skipping the day not the instance of completing the task.

For the streak issue you need to be a bit more specific for us to understand what's happening (I have a 2 per month habit that gets streaks). Which days have you completed?

**Reply by pseudacris** [reply]
The 2 times per week (Sunday - Saturday) habit is binary and it doesn't matter which 2 days. The 2 times a month habit is exactly the same other than scale. Both have the amount of days required but no streak, like it expects me to enter more days. It counts each week in the 2 day habit as a separate streak unless I use skip days. The 2 times a month habit never showed any kind of streak despite having 2 days within a calendar month entered, which is why I ended up deleting it and using another app, but I would prefer to just use Loop.

**Reply by KyleSCraig** [reply]
Here's the type of answer I was hoping for, though also a screenshot could be an easier way to convey some of the information:

I make a habit with 2 times per week. I tick Saturday 9 November, Sunday 10 November, Saturday 16 November, Sunday 17 November. In addition to the habits completed, there are the following dates as autocompleted (tick outline, lightly shaded in calendar view): Every other date from Monday 4 November to Friday 22 November (inclusive of both). The streak is 19 from Nov 4, 2024 to Nov 22, 2024. No dates are skipped, (which I know is different from what you described).

I make a habit with 2 times per month. I tick Saturday 2 November and Sunday 17 November. All other dates from 2 November to 1 December are autocompleted, so there's a 30 day streak from Nov 2, 2024 to Dec 1, 2024.



If I understand you right, you're saying if you tick the dates I have said, then you get 0 autocompletes. In that case I would suggest as an investigation, tick dates subsequently (So Monday 18 November, etc...) and see if you can get any autocompletes. If you don't, what happens if you change it down to 1 per month/week? Does it work like a normal 1 per month/week?

**Reply by pseudacris** [reply]
Sorry, I didn't realize I could add images. Thanks for your patience. 

Here's pictures of the habit's conditions, the habit without a skip day, and the habit with a skip day. I get allergy shots twice a week and those days vary week to week. Both weeks already have 2 shots each, meeting the requirement I set for it of 2 instances per week. Yet if I don't put a skip day at the beginning of the second week it doesn't count it as a 15 day streak, it counts it as two 7 day streaks. I was under the impression that the streak would continue uninterrupted week after week (Sunday thru Saturday) as long as there are 2 days that I mark with the specific days not mattering. 


![Screenshot_20241205-163208.png](https://github.com/user-attachments/assets/ebb6dd41-d84e-48f2-999f-cf79a352f596)

![Screenshot_20241205-163117.png](https://github.com/user-attachments/assets/d45a4000-9348-414d-83e1-9c3d77d2a802)

![Screenshot_20241205-163105.png](https://github.com/user-attachments/assets/2d67d4b0-cb0f-43ad-a164-c1f19b6ee54f)

**Reply by KyleSCraig** [reply, marked as answer]
Okay, yes I do have that behaviour too.

I believe actually what you need is #1851 and #1999, and it's not a change I'm confident about making. Correct me if I'm wrong, but if you use "Hide completed" (which I highly recommend normally), then even if you do put the skip in, your streak will continue to the Monday 9th, when it shouldn't autocomplete because you might be supposed to take it on that day. Unfortunately my best workaround suggestion is to input the first occasion on Monday and the second occasion on Tuesday - regardless of what the actual days were.

I will point out that if you do enter a completion on Monday 9th, then it will backfill and connect up your streak without need for skips. So after 10 weeks of perfect completion, you'll have a big streak, and then depending on the day you might have break before a new streak, but eventually that will get connected back up to the main one.

**Reply by pseudacris** [reply]
Thanks for the workaround, I'll do that or use dashes for now. Looks like this has been a feature request for awhile, hopefully it gets added. I'm glad I understand how it works and it's not just on my end. I appreciate your time.
